funsor/pyro/convert.py

```diff
@@ -65,28 +67,47 @@ def funsor_to_tensor(funsor_, ndims, event_inputs=()):
     return tensor
 
 
+@singledispatch
 def dist_to_funsor(pyro_dist, event_inputs=()):
     """
     Convert a :class:`torch.distributions.Distribution` to a
     :class:`~funsor.terms.Funsor` .
     """
     assert isinstance(pyro_dist, torch.distributions.Distribution)
+    raise ValueError("Cannot convert {} distribution to a Funsor"
+                     .format(type(pyro_dist).__name__))
```

Comment:
Do you think it would be possible to use the `torch.distributions.Distribution.param_shape` attribute to handle `dist_to_funsor` conversion generically for the subset of distributions that implement it?

Reply:
Hmm, I'd have to look at more special cases to generalize.

Reply:
Ok, no need to address it in this PR, just something to keep in mind